modules/bridgewellBidAdapter.js

```diff
@@ -0,0 +1,266 @@
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, NATIVE],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    let valid = false;
+
+    if (bid && bid.params && bid.params.ChannelID) {
+      valid = true;
+    }
+
+    return valid;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} validBidRequests - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const adUnits = [];
+    utils._each(validBidRequests, function (bid) {
+      adUnits.push({
+        ChannelID: bid.params.ChannelID,
+        mediaTypes: bid.mediaTypes || {
+          banner: {
+            sizes: bid.sizes
+          }
+        }
+      });
+    });
+
+    let topUrl = '';
+    if (bidderRequest && bidderRequest.refererInfo) {
+      topUrl = bidderRequest.refererInfo.referer;
+    }
+
+    return {
+      method: 'POST',
+      url: REQUEST_ENDPOINT,
+      data: {
+        version: {
+          prebid: '$prebid.version$',
+          bridgewell: BIDDER_VERSION
+        },
+        inIframe: utils.inIframe(),
+        url: topUrl,
+        referrer: getTopWindowReferrer(),
+        adUnits: adUnits
+      },
+      validBidRequests: validBidRequests
+    };
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @param {*} bidRequest
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const bidResponses = [];
+
+    // map responses to requests
+    utils._each(bidRequest.validBidRequests, function (req) {
+      const bidResponse = {};
+
+      if (!serverResponse.body) {
+        return;
+      }
+
+      let matchedResponse = find(serverResponse.body, function (res) {
+        let valid = false;
+
+        if (res) {
+          let mediaTypes = req.mediaTypes;
+          if (res.width && res.height && mediaTypes) {
+            if (mediaTypes.native) { // dont care native sizes
+              valid = true;
+            } else if (mediaTypes.banner) {
+              if (mediaTypes.banner.sizes) {
+                let width = res.width;
+                let height = res.height;
+                let sizes = mediaTypes.banner.sizes;
+                // check response size validation
+                if (typeof sizes[0] === 'number') { // for foramt Array[Number] check
+                  valid = width === sizes[0] && height === sizes[1];
+                } else { // for format Array[Array[Number]] check
+                  valid = find(sizes, function (size) {
```

Comment:
`valid` will be set equal to the element returned from `find`. Should be a boolean instead?

Reply:
fixed, thank you.